File: test/commands/view.test.js
import {
  runClangdQuery,
  assert,
  assertContains,
  assertMatches,
  countOccurrences,
  waitForDaemonReady,
} from '../test-helpers.js';

console.log('Testing view command...\n');

// Ensure daemon is ready before running tests
console.log('Waiting for daemon to be ready...');
await waitForDaemonReady();

// Test 1: View a complete class
console.log('Test 1: View GameObject class');
const result1 = await runClangdQuery(['view', 'GameObject']);
assert(result1.exitCode === 0, 'Command should succeed');
// Should show the complete class definition
assertContains(result1.stdout, 'class GameObject : public Updatable');
assertContains(result1.stdout, 'void Update(float delta_time) override');
assertContains(result1.stdout, 'Transform transform_;');
assertContains(result1.stdout, 'std::vector<std::shared_ptr<Component>> components_;');
console.log('✓ Test 1 passed\n');

// Test 2: View a specific method
console.log('Test 2: View specific method GameObject::Update');
const result2 = await runClangdQuery(['view', 'GameObject::Update']);
assert(result2.exitCode === 0, 'Command should succeed');
// Should show the method implementation
assertContains(result2.stdout, 'void GameObject::Update(float delta_time)');
assertContains(result2.stdout, 'OnUpdate(delta_time);');
console.log('✓ Test 2 passed\n');

// Test 3: View a regular class (Factory)
console.log('Test 3: View Factory class');
const result3 = await runClangdQuery(['view', 'Factory']);
assert(result3.exitCode === 0, 'Command should succeed');
// Should show the Factory class
assertContains(result3.stdout, 'class Factory');
assertContains(result3.stdout, 'std::unique_ptr<Base> Create(const std::string& type_name)');
assertContains(result3.stdout, 'void Register(const std::string& type_name, Creator creator)');
console.log('✓ Test 3 passed\n');

// Test 4: View non-existent symbol
console.log('Test 4: View non-existent symbol');
const result4 = await runClangdQuery(['view', 'NonExistentClass']);
assert(result4.exitCode === 0, 'Command should succeed even with no results');
assertContains(result4.stdout, 'No symbols found matching "NonExistentClass"');
console.log('✓ Test 4 passed\n');

console.log('All view tests passed! ✓');

File: go/test/view_test.go
package test

import (
	"testing"
)

func TestViewCommand(t *testing.T) {
	tc := GetTestContext(t)
	
	t.Run("View GameObject complete class", func(t *testing.T) {
		result := tc.RunCommand("view", "GameObject")
		tc.AssertExitCode(result, 0)
		// Should show the complete class definition
		tc.AssertContains(result.Stdout, "class GameObject : public Updatable")
		tc.AssertContains(result.Stdout, "void Update(float delta_time) override")
		tc.AssertContains(result.Stdout, "Transform transform_;")
		tc.AssertContains(result.Stdout, "std::vector<std::shared_ptr<Component>> components_;")
	})
	
	t.Run("View specific method GameObject::Update", func(t *testing.T) {
		result := tc.RunCommand("view", "GameObject::Update")
		tc.AssertExitCode(result, 0)
		// Should show the method implementation
		tc.AssertContains(result.Stdout, "void GameObject::Update(float delta_time)")
		tc.AssertContains(result.Stdout, "OnUpdate(delta_time);")
	})
	
	t.Run("View Factory class", func(t *testing.T) {
		result := tc.RunCommand("view", "Factory")
		tc.AssertExitCode(result, 0)
		// Should show the Factory class
		tc.AssertContains(result.Stdout, "class Factory")
		tc.AssertContains(result.Stdout, "std::unique_ptr<Base> Create(const std::string& type_name)")
		tc.AssertContains(result.Stdout, "void Register(const std::string& type_name, Creator creator)")
	})
	
	t.Run("View non-existent symbol", func(t *testing.T) {
		result := tc.RunCommand("view", "NonExistentClass")
		tc.AssertExitCode(result, 0)
		tc.AssertContains(result.Stdout, "No symbols found matching \"NonExistentClass\"")
	})
}